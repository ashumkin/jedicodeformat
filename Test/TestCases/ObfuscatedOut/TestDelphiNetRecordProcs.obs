unit testdelphinetrecordprocs;interface type tsomerecord=record i,j,k,l:integer;
function sum:integer;procedure clear;constructor create(ivalue:integer);end;toprecord
=record i:integer;s:string;class operator add(a,b:toprecord):toprecord;class operator
equal(a,b:toprecord):boolean;class operator implicit(a:toprecord):integer;class operator
implicit(a:integer):toprecord;end;tpropertyrecord=record private fi,fj:integer;function
gettwicei:integer;function gettwicej:integer;procedure settwicei(const value:integer
);public property i:integer read fi write fi;property twicei:integer read gettwicei
write settwicei;property twicej:integer read gettwicej;end;implementation procedure
tsomerecord.clear;begin i:=0;j:=0;k:=0;l:=0;end;constructor tsomerecord.create(ivalue
:integer);begin clear;i:=ivalue;end;function tsomerecord.sum:integer;begin result
:=i+j+k+l;end;class operator toprecord.add(a,b:toprecord):toprecord;begin result.
i:=a.i+b.i;result.s:=a.s+b.s;end;class operator toprecord.equal(a,b:toprecord):boolean
;begin result:=(a.i=b.i)and(a.s=b.s);end;class operator toprecord.implicit(a:integer
):toprecord;begin result.i:=a;result.s:='';end;class operator toprecord.implicit(
a:toprecord):integer;begin result:=a.i;end;function tpropertyrecord.gettwicei:integer
;begin result:=fi*2;end;function tpropertyrecord.gettwicej:integer;begin result:=
fj*2;end;procedure tpropertyrecord.settwicei(const value:integer);begin fi:=value
div 2;end;end. 