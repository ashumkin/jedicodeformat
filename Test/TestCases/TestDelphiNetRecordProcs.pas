unit TestDelphiNetRecordProcs;

interface

type
  TSomeRecord = record
    i,j,k,l: integer;

    function Sum: integer;
    procedure Clear;

    constructor Create(iValue: integer);
  end;

implementation


{ TSomeRecord }

procedure TSomeRecord.Clear;
begin
  i := 0;
  j := 0;
  k := 0;
  l := 0;
end;

constructor TSomeRecord.Create(iValue: integer);
begin
  Clear;
  i := iValue;
end;

function TSomeRecord.Sum: integer;
begin
  Result := i + j + k + l;
end;

end.
