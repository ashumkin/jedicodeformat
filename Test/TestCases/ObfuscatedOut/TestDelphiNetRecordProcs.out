unit testdelphinetrecordprocs;

interface

type
  tsomerecord = record
    i, j, k, l: integer;
    function sum: integer;
    procedure Clear;
    constructor Create(ivalue: integer);
  end;

  toprecord = record
    i: integer;
    s: string;
    class operator add(a, b: toprecord): toprecord;
    class operator
      equal(a, b: toprecord): boolean;
    class operator implicit(a: toprecord): integer;
    class operator
      implicit(a: integer): toprecord;
  end;
  tpropertyrecord = record
  private
    fi, fj: integer;
    function gettwicei: integer;
    function gettwicej: integer;
    procedure settwicei(const Value: integer);
  public
    property i: integer Read fi Write fi;
    property twicei: integer Read gettwicei Write settwicei;
    property twicej: integer Read gettwicej;
  end;

implementation

procedure tsomerecord.Clear;
begin
  i := 0;
  j := 0;
  k := 0;
  l := 0;
end;

constructor tsomerecord.Create(ivalue: integer);
begin
  Clear;
  i := ivalue;
end;

function tsomerecord.sum: integer;
begin
  Result := i + j + k + l;
end;

class operator toprecord.add(a, b: toprecord): toprecord;
begin
  Result.
    i      := a.i + b.i;
  Result.s := a.s + b.s;
end;

class operator toprecord.equal(a, b: toprecord): boolean;
begin
  Result := (a.i = b.i) and (a.s = b.s);
end;

class operator toprecord.implicit(a: integer): toprecord;
begin
  Result.i := a;
  Result.s := '';
end;

class operator toprecord.implicit(a: toprecord): integer;
begin
  Result := a.i;
end;

function tpropertyrecord.gettwicei: integer;
begin
  Result := fi * 2;
end;

function tpropertyrecord.gettwicej: integer;
begin
  Result :=
    fj * 2;
end;

procedure tpropertyrecord.settwicei(const Value: integer);
begin
  fi := Value div 2;
end;

end.
 